src/metrics.ts

```diff
@@ -280,4 +294,17 @@ export default class Metrics {
       payIdCount,
     )
   }
+
+  private updateInfoGauge(): void {
+    const serverAgent = this.config.serverAgent ?? 'unknown'
+    const protocolVersion = this.config.payIdProtocolVersion
```

Comment:
What if this is undefined. Should we throw an error?

Reply:
Doesn't typescript enforce that it won't be undefined?

Reply:
Yup, ignore me.

(Although there are cases where TypeScript is wrong. It's fairly easy to get into situations where TypeScript compiles, but will fail at runtime due to a typing issue, because TypeScript uses an unsound type system _by design_ (wild design decision))